pkg/scheduler/util/utils.go

```diff
@@ -119,6 +122,32 @@ func PatchPodStatus(ctx context.Context, cs kubernetes.Interface, old *v1.Pod, n
 	return err
 }
 
+// AddConditionAndDeletePod prepares and deletes the given <pod> from API server
+func AddConditionAndDeletePod(ctx context.Context, cs kubernetes.Interface, pod *v1.Pod, podConditionType v1.PodConditionType) error {
+	if err := AddPodCondition(ctx, cs, pod, podConditionType); err != nil {
+		return err
+	}
+	return DeletePod(ctx, cs, pod)
+}
+
+// AddPodCondition prepares pod for deletion - add a dedicated condition indicating the reason for deletion
+func AddPodCondition(ctx context.Context, cs kubernetes.Interface, pod *v1.Pod, podConditionType v1.PodConditionType) error {
+	if feature.DefaultFeatureGate.Enabled(features.JobBackoffPolicy) {
+		return patchPodStatusWithCondition(ctx, cs, pod, podConditionType)
+	}
+	return nil
+}
+func patchPodStatusWithCondition(ctx context.Context, cs kubernetes.Interface, pod *v1.Pod, podConditionType v1.PodConditionType) error {
+	podStatusCopy := pod.Status.DeepCopy()
+	if podutil.UpdatePhaseAndAddConditionForFailedPod(podStatusCopy, &v1.PodCondition{
```

Comment:
Preemption shouldn't set the phase to Failed. The pod didn't finish yet. It's responsibility of the kubelet to do that.

We can only add the phase when we know that the kubelet is down, for example when the pod is orphan or the node is unreacheable.